### DISCUSSION TITLE
Stacker with multiple surfaces-based bodies does not work

### DISCUSSION THREAD

**Discussion by giovannipappalardiansys** [ansys/pyprimemesh#668] [Q&A] [answered]
### 🔍 Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### 🐞 Description of the bug

If I have 1 body the stacker works

If I have two bodies stacker does not work.

### 📝 Steps to reproduce

here relevant files: CAD, script and Slides that show the issue

### 💻 Which operating system are you using?

Windows

### 📀 Which ANSYS version are you using?

2023R2

### 🐍 Which Python version are you using?

3.10

### 📦 Installed packages

```shell
PyPrimeMesh
```

**Comment by RagulSEL** [comment]
Please check that, the topology between multiple bodies there or not. If yes, please check the stacker orientation once

**Comment by waltersma** [comment, marked as answer]
Hi @giovannipappalardiansys, 

This looks to be due to the lateral defeaturing tolerance you are using for creating the base face.  I see it is creating several more faces than I believe you are expecting due to the small tolerance it is picking up the variation in the curved edges between the upper and lower circular edges and creating very small sliver faces that fail to mesh successfully without cleanup.  

I would suggest using a value in the region of 20% of your mesh size to avoid this for this case. Arbitrarily I used a value of 0.1 and this worked for me.

Hope this helps! Thanks

**Comment by giovannipappalardiansys** [comment]
@waltersma 
this solved the issues I have!

Thanks a lot!
